03/GeoffRiley/tags.py

```diff
@@ -0,0 +1,41 @@
+import re
+from collections import Counter
+from difflib import SequenceMatcher
+from itertools import permutations
+
+TOP_NUMBER = 10
+RSS_FEED = 'rss.xml'
+SIMILAR = 0.87
+
+TAG_REGEX = re.compile(r'<category>([^<]+?)</category>')
```

Comment:
if `[^<]+?` should match anything that is not a tag thats ok but if you use `(.*?)` this is more common as the ? means "not greedy version" and the closing </category> will mark the end. Without the ? you are right that the greedy version of the dot notation will match even the closing tag and beyond.

Reply:
I spent 30 min debugging a submission yesterday where there was just one month / int wrong due to a greedy/non-greedy problem :)

Reply:
My reason for using `[^<]+?` rather than `.*?` is to emphasise that it is a second tag instance that is being avoided.  It also automatically excludes any empty tags.